singularity-eos/eos/singularity_eos.cpp

```diff
@@ -19,6 +19,34 @@
 #include <singularity-eos/eos/singularity_eos.hpp>
 #include <singularity-eos/eos/singularity_eos_init_utils.hpp>
 
+/*
+===============================================
+2D Lambda indexer class. Usage is as follows:
+
+Assuming lambda is a std::array<Real, nCell>, instantiate this class
+as
+
+   idx = lambdaIndexer2D(lambda.data(), n)
+
+We can now use the [] operator as follows:
+
+   idx[i]
+
+which will return the memory address of the element n*i of the array lambda
+===============================================
+*/
+
+class lambdaIndexer2D {
+ public:
+  lambdaIndexer2D(int n, double *data) : n_(n), data_(data) {}
+
+  double *operator[](int i) const { return &(data_[n_ * i]); }
```

Comment:
This needs device markings, which is why tests are failing on re-git
```suggestion
  PORTABLE_FORCEINLINE_FUNCTION
  double *operator[](int i) const { return &(data_[n_ * i]); }
```

Reply:
> Added

The re-git CI is still failing. Not sure what the error means